packages/api-form-builder/src/plugins/crud/forms.crud.ts

```
@@ -305,15 +268,16 @@ export const createFormsCrud = (params: CreateFormsCrudParams): FormsCRUD => {
                 webinyVersion: context.WEBINY_VERSION
             };
 
+            let result;
```

We should create the `formStats` entry only if the `form` entry is successfully created. 

Right now, we are trying to create the `formStats` entry even when the creation fails, right?

---

The code inside the try-catch will throw an error if the form is not created. This prevents the `formStats` section from being executed. The same applies to the next two comments as well.

---

Ok, let's throw a meaningful error in case the `formStats` creation fails, something like:

```
throw new WebinyError(
    ex.message || "Could not create form.",
    ex.code || "CREATE_FORM_STATS_ERROR",
    {
        form
    }
);

---

Errors are already implemented for all formStats CRUD operations.

In the `createForm` method, if form creation fails, it will throw a [CREATE_FORM_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/forms.crud.ts#L282). If it succeeds, the `createFormStats` method is executed, which, in case of failure, will throw a [CREATE_FORM_STATS_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/formStats.crud.ts#L154).

Similarly, in the `createFormRevision` method, if form revision creation fails, a [CREATE_FORM_FROM_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/forms.crud.ts#L530) is thrown. However, if the form revision is successfully created but `createFormStats` fails, a [CREATE_FORM_STATS_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/formStats.crud.ts#L154) is thrown.

Lastly, for `deleteForm`, if form deletion fails, it throws a [DELETE_FORM_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/forms.crud.ts#L374). Conversely, if the form is deleted successfully but `deleteFormStats` encounters an error, a [DELETE_FORM_STATS_ERROR](https://github.com/webiny/webiny-js/blob/neatbyte/form-stats/packages/api-form-builder/src/plugins/crud/formStats.crud.ts#L220) is thrown.